### PR TITLE
Introduce multi-endpoint servers and health check endpoints to HttpServerSourceStage

### PR DESCRIPTION
**Pull request by jadu-nv**
## Description

Addresses: https://github.com/nv-morpheus/Morpheus/issues/1732

Currently, the `HttpServer` implementation does not support the initialization of a single server with multiple endpoints. This PR seeks to address this issue by refactoring the `HttpServer` by taking a list of `HttpEndpoint` objects (encapsulates the attributes of an http endpoint, ie. url, method, endpoint handler).

Additionally, this PR also introduces 2 endpoints to `HttpServerSourceStage` that allows users to conduct health checks on their pipelines.
- GET /live -> Immediately returns a 200 response if the server is alive
- GET /ready -> Immediately returns a 200 response if the server is ready to accept new messages

These endpoints are configurable by the user.

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: morpheus/_lib/src/utilities/http_server.cpp [hunk 1]
```diff
@@ -127,18 +123,24 @@ class Session : public std::enable_shared_from_this<Session>
     {
         DLOG(INFO) << "Received request: " << request.method() << " : " << request.target();
         m_response = std::make_unique<http::response<http::string_body>>();
+        bool valid_request = false;
 
-        if (request.target() == m_url_endpoint && (request.method() == m_method))
+        for (const auto& endpoint : m_endpoints)
```
Comment (dagardner-nv) [inline morpheus/_lib/src/utilities/http_server.cpp:128]: I would replace the range loop with an old-style loop allowing the loop to exit early once `valid_request` is `true`

File: morpheus/_lib/src/utilities/http_server.cpp [hunk 2]
```diff
@@ -127,18 +123,24 @@ class Session : public std::enable_shared_from_this<Session>
...
+            if (request.target() == endpoint.m_url && request.method() == endpoint.m_method)
+            {
+                valid_request = true;
+                std::string body{request.body()};
+                auto parse_status = (*endpoint.m_parser)(body);
+
+                m_response->result(std::get<0>(parse_status));
+                m_response->set(http::field::content_type, std::get<1>(parse_status));
+                m_response->body() = std::get<2>(parse_status);
+                m_on_complete_cb   = std::get<3>(parse_status);
```
Comment (mdemoret-nv) [inline morpheus/_lib/src/utilities/http_server.cpp:139]: Add a `break` here to exit the loop

**Review by mdemoret-nv — APPROVED** [review]

File: morpheus/_lib/include/morpheus/utilities/http_server.hpp [hunk 3]
```diff
@@ -110,18 +110,30 @@ class MORPHEUS_EXPORT HttpServer
...
-    std::shared_ptr<payload_parse_fn_t> m_payload_parse_fn;
     std::atomic<bool> m_is_running;
 };
 
+/**
+ * @brief A struct that encapsulates the http endpoint attributes
+ *
+ * @details Constructed to be used in the HttpServer class as http endpoint configurations
+ */
+struct MORPHEUS_EXPORT HttpEndpoint
```
[the remark this answers is not in the document]
Comment (jadu-nv) [reply morpheus/_lib/include/morpheus/utilities/http_server.hpp:128]: To maintain cohesiveness, should I also move the Listener class declaration to be in the same area and remove its forward-declaration?

[the remark this answers is not in the document]
Comment (dagardner-nv) [reply morpheus/_lib/include/morpheus/utilities/http_server.hpp:128]: Yes that makes sense

File: morpheus/_lib/src/utilities/http_server.cpp (hunk 1 above)
Comment (jadu-nv) [reply morpheus/_lib/src/utilities/http_server.cpp:128]: Added `break` statement instead

File: morpheus/_lib/include/morpheus/utilities/http_server.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jadu-nv) [reply morpheus/_lib/include/morpheus/utilities/http_server.hpp:79]: Removed forward-declarations

File: morpheus/_lib/src/utilities/http_server.cpp (hunk 2 above)
Comment (jadu-nv) [reply morpheus/_lib/src/utilities/http_server.cpp:139]: Added a `break` statement here

**Pull request merged**